src/js/components/form-elements/v2/FileSelect.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react';
+
+import PropTypes from 'prop-types';
+import { useDropzone } from 'react-dropzone';
+
+import Button from 'components/form-elements/Button';
+import useTranslate from 'hooks/useTranslate';
+
+import './style.scss';
+
+const FileSelect = ({
+  height,
+  width,
+  dropzoneText,
+  buttonLabel,
+  className,
+  multiple,
+  maxFiles,
+  allowedExtensions,
+  ...fieldProps
+}) => {
+  const onDrop = useCallback((acceptedFiles) => {
+    fieldProps.onChange?.(multiple ? acceptedFiles : acceptedFiles[0]);
+  }, []);
+
+  const translate = useTranslate();
+
+  const getFileExtension = (file) => file.path.split('.')?.[1];
+
+  const validateFileType = (file) => {
+    if (!allowedExtensions.length || allowedExtensions.includes(getFileExtension(file))) {
+      return null;
```

Comment:
I think for clarity of validation methods we used to have, this should eventually return `true` if the validation passes, which I think it does in this case if I understand it correctly:
"If we don't have any allowed extensions specified or the file extension is in the allowed ones, the validation should pass (currently returning null)"?

Reply:
The `null` is required by the `react-dropzone`. Basically, this validation blocks you from uploading files with the wrong extensions.